php-frontend/src/main/java/org/sonar/php/parser/PHPGrammar.java

```diff
@@ -966,22 +966,31 @@ public InternalSyntaxToken EOS() {
 
   public ExpressionTree CAST_EXPR() {
     return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.CAST_TYPE).is(
-      f.castExpression(
-        b.token(PHPPunctuator.LPARENTHESIS),
-        b.firstOf(
-          b.token(PHPKeyword.ARRAY),
-          b.token(PHPKeyword.UNSET),
-          b.token(PHPLexicalGrammar.INTEGER),
-          b.token(PHPLexicalGrammar.INT),
-          b.token(PHPLexicalGrammar.DOUBLE),
-          b.token(PHPLexicalGrammar.FLOAT),
-          b.token(PHPLexicalGrammar.STRING),
-          b.token(PHPLexicalGrammar.OBJECT),
-          b.token(PHPLexicalGrammar.BOOLEAN),
-          b.token(PHPLexicalGrammar.BOOL),
-          b.token(PHPLexicalGrammar.BINARY)),
-        b.token(PHPPunctuator.RPARENTHESIS),
-        UNARY_EXPR()));
+      b.firstOf(
+        f.castExpression(
+          b.token(PHPPunctuator.LPARENTHESIS),
+          b.firstOf(
+            b.token(PHPKeyword.ARRAY),
+            b.token(PHPKeyword.UNSET),
+            b.token(PHPLexicalGrammar.INTEGER),
+            b.token(PHPLexicalGrammar.INT),
+            b.token(PHPLexicalGrammar.DOUBLE),
+            b.token(PHPLexicalGrammar.FLOAT),
+            b.token(PHPLexicalGrammar.STRING),
+            b.token(PHPLexicalGrammar.OBJECT),
+            b.token(PHPLexicalGrammar.BOOLEAN),
+            b.token(PHPLexicalGrammar.BOOL),
+            b.token(PHPLexicalGrammar.BINARY)),
+          b.token(PHPPunctuator.RPARENTHESIS),
+          UNARY_EXPR()),
+        f.prefixedCastExpression(
```

Comment:
why not making a rule from this directly?
It would be easier to read `PREFIXED_BINARY_CAST()` here.